qiskit_experiments/framework/experiment_data.py

```diff
@@ -1322,24 +1408,48 @@ def _retrieve_analysis_results(self, refresh: bool = False):
                 an experiment service is available.
         """
         # Get job results if missing experiment data.
-        if self.service and (not self._analysis_results or refresh):
+        if self.service and (len(self._analysis_results) == 0 or refresh):
             retrieved_results = self.service.analysis_results(
                 experiment_id=self.experiment_id, limit=None, json_decoder=self._json_decoder
             )
             for result in retrieved_results:
-                result_id = result.result_id
-
-                self._analysis_results[result_id] = AnalysisResult(service=self.service)
-                self._analysis_results[result_id].set_data(result)
-                self._analysis_results[result_id]._created_in_db = True
+                # Canonicalize IBM specific data structure.
+                # TODO define proper data schema on frontend and delegate this to service.
+                cano_quality = AnalysisResult.RESULT_QUALITY_TO_TEXT.get(result.quality, "unknown")
+                cano_components = [to_component(c) for c in result.device_components]
+                extra = result.result_data["_extra"]
+                if result.chisq is not None:
+                    extra["chisq"] = result.chisq
+                self._analysis_results.add_entry(
+                    name=result.result_type,
+                    value=result.result_data["_value"],
+                    quality=cano_quality,
+                    components=cano_components,
+                    experiment_id=result.experiment_id,
+                    result_id=result.result_id,
+                    tags=result.tags,
+                    backend=result.backend_name,
+                    created_time=result.creation_datetime,
+                    **extra,
+                )
 
+    @deprecate_arg(
+        name="dataframe",
+        deprecation_description="Setting ``dataframe`` to False in analysis_results",
+        since="0.6",
+        package_name="qiskit-experiments",
+        pending=True,
+        predicate=lambda dataframe: not dataframe,
+    )
     def analysis_results(
         self,
         index: Optional[Union[int, slice, str]] = None,
         refresh: bool = False,
         block: bool = True,
         timeout: Optional[float] = None,
-    ) -> Union[AnalysisResult, List[AnalysisResult]]:
+        verbosity: int = 2,
```

Comment:
I'd prefer more explicit interface accepting `columns: Union[str, List[str]] ="DEFAULT"` directly with some special column names e.g. "MINIMAL"/"DEFAULT"/"ALL" that is equivalent with verbosity=1/2/3: 
```
analysis_results(dataframe=True, columns=["name", "value"])
analysis_results(dataframe=True, columns="ALL")
```

Reply:
Thanks for suggestion. Done in [0d5f4ae](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1133/commits/0d5f4ae1e546c74a4e2485a686b08dd54c605967)